engine/src/operators/GroupBy.cpp

```diff
@@ -732,37 +691,54 @@ void process_aggregate(blazing_frame & input, std::string query_part, Context *
 		}
 	}
 
+	std::unique_ptr<ral::frame::BlazingTable> ret;
+	
 	if(aggregation_types.size() == 0) {
 		if(!queryContext || queryContext->getTotalNodes() <= 1) {
-			single_node_groupby_without_aggregations(input, group_column_indices);
+			ret = single_node_groupby_without_aggregations(table, group_column_indices);
```

Comment:
only have one version that handles both, handle that logic inside that function by passing it the context